### PR TITLE
Adding compatibility for marathon 1.5

### LINKED ISSUE #2115: Marathon 1.5.0 breaks Traefik Marathon integration
### Do you want to request a *feature* or report a *bug*?

Report a bug.

### What did you do?

Recently upgraded DCOS cluster from 1.9 to 1.10, which includes an upgrade to Marathon 1.5.0-SNAPSHOT-713-g14280a6.

### What did you expect to see?
Backends generated by marathon plugin use IP-per-task IP. For the service below, expected the backend to be set to 9.0.6.133:80.
```json
{
    "app": {
        "backoffFactor": 1.15,
        "backoffSeconds": 1,
        "container": {
            "docker": {
                "forcePullImage": false,
                "image": "<redacted>",
                "parameters": [],
                "privileged": false
            },
            "portMappings": [
                {
                    "containerPort": 80,
                    "labels": {},
                    "name": "http",
                    "protocol": "tcp",
                    "servicePort": 10107
                }
            ],
            "type": "DOCKER",
            "volumes": []
        },
        "cpus": 0.2,
        "deployments": [],
        "disk": 0,
        "executor": "",
        "gpus": 0,
        "healthChecks": [
            {
                "delaySeconds": 15,
                "gracePeriodSeconds": 300,
                "ignoreHttp1xx": false,
                "intervalSeconds": 60,
                "maxConsecutiveFailures": 3,
                "path": "/",
                "portIndex": 0,
                "protocol": "HTTP",
                "timeoutSeconds": 20
            }
        ],
        "id": "/internal-ticketing",
        "instances": 1,
        "killSelection": "YOUNGEST_FIRST",
        "labels": {<Redacted>},
        "maxLaunchDelaySeconds": 3600,
        "mem": 512,
        "networks": [
            {
                "mode": "container",
                "name": "dcos"
            }
        ],
        "requirePorts": false,
        "tasks": [
            {
                "appId": "/internal-ticketing",
                "healthCheckResults": [
                    {
                        "alive": true,
                        "consecutiveFailures": 0,
                        "firstSuccess": "2017-09-12T21:18:31.612Z",
                        "instanceId": "internal-ticketing.marathon-ea90760d-97ff-11e7-8013-629e7338b525",
                        "lastFailure": null,
                        "lastFailureCause": null,
                        "lastSuccess": "2017-09-13T14:24:52.122Z"
                    }
                ],
                "host": "10.172.130.189",
                "id": "internal-ticketing.ea90760d-97ff-11e7-8013-629e7338b525",
                "ipAddresses": [
                    {
                        "ipAddress": "9.0.6.133",
                        "protocol": "IPv4"
                    }
                ],
                "ports": [],
                "slaveId": "c26773ac-c1e8-44d2-a012-4bb3159e1762-S3",
                "stagedAt": "2017-09-12T21:18:26.589Z",
                "startedAt": "2017-09-12T21:18:29.319Z",
                "state": "TASK_RUNNING",
                "version": "2017-09-12T21:18:26.559Z"
            }
        ],
        "tasksHealthy": 1,
        "tasksRunning": 1,
        "tasksStaged": 0,
        "tasksUnhealthy": 0,
        "unreachableStrategy": {
            "expungeAfterSeconds": 600,
            "inactiveAfterSeconds": 300
        },
        "upgradeStrategy": {
            "maximumOverCapacity": 1,
            "minimumHealthCapacity": 1
        },
        "version": "2017-09-12T21:18:26.559Z",
        "versionInfo": {
            "lastConfigChangeAt": "2017-09-12T21:18:26.559Z",
            "lastScalingAt": "2017-09-12T21:18:26.559Z"
        }
    }
}
```

### What did you see instead?
Instead the backend is created using the host IP: 10.172.130.189:80

### Output of `traefik version`: (_What version of Traefik are you using?_)

(Running in container based off of roquefort-alpine)
```
Version:      v1.4.0-rc2
Codename:     roquefort
Go version:   go1.9
Built:        2017-09-08_07:31:37PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

No arguments. Platform is DCOS 1.10 running in AWS.

```toml
defaultEntryPoints = ["http","https"]

[web]

# Web administration port
#
# Required
#
address = ":8080"

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
  [entryPoints.es]
  address = ":10020"
[marathon]
watch = true
domain = "marathon.mesos"
forceTaskHostname = false
exposedByDefault = true
marathonLBCompatibility = true
endpoint = "http://marathon.mesos:8080"
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

N/A

### PR DESCRIPTION
**Pull request by TrevinTeacutter**
### What does this PR do?

Committing changes we made in our private repo to support Marathon 1.5 in Traefik.
We have been using this with marathon 1.5 for since Traefik 1.5 (this also supports Marathon 1.4).

I have not been able to test that the backwards compatibility since we rebased these changes with 1.6 of Traefik.

Updating go-marathon is a requirement for this.

### Motivation

Fixes #2115

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

**Comment by mmatur** [comment]
@containous/marathon WDYT ?

**Comment by TrevinTeacutter** [comment]
Working on trying to get integration tests, but getting the containers to play nice since Marathon 1.5 increased the mesos version requirements is being a bit more painful that I expected.

For documentation, is there anything I need to add? Not sure whether there was anything around the fact that only Marathon <=1.4 is supported.

**Comment by timoreimann** [comment]
@TrevinTeacutter

>For documentation, is there anything I need to add? Not sure whether there was anything around the fact that only Marathon <=1.4 is supported.

I think the most important aspect we should check is whether there are any Marathon-specific parameters for which the behavior has changed with 1.5. Specifically, I'm wondering about `forceTaskHostname`.

Would you mind going over the Marathon backend documentation and the Marathon user guide and verify if anything might not be correct with 1.5 anymore?

**Comment by TrevinTeacutter** [comment]
The only non-passive change that I know of was around networking related structures. I went over all the documentation and didn't notice anything that should be broken with 1.5 other than routing information.

File: integration/marathon_test.go [hunk 1]
```diff
@@ -138,3 +138,51 @@ func (s *MarathonSuite) TestConfigurationUpdate(c *check.C) {
 	err = try.GetRequest("http://127.0.0.1:8000/app", 30*time.Second, try.StatusCodeIs(http.StatusOK))
 	c.Assert(err, checker.IsNil)
 }
+
+// func (s *MarathonSuite) TestHostNetwork(c *check.C) {
```
Comment (Juliens) [inline integration/marathon_test.go]: Could you remove this method or uncomment it?

**Pull request merged**